engine/lib/generic_printer/generic_printer.ml

```diff
@@ -231,14 +231,15 @@ module Make (F : Features.T) = struct
       [local] is true, otherwise it prints the full path, separated by
       `module_path_separator`. *)
 
-      method quote (quote : quote) : document =
+      method quote ~contents ~witness:_ : document =
         List.map
-          ~f:(function
-            | `Verbatim code -> string code
-            | `Expr e -> self#print_expr AstPosition_Quote e
-            | `Pat p -> self#print_pat AstPosition_Quote p
-            | `Typ p -> self#print_ty AstPosition_Quote p)
-          quote.contents
+          ~f:(fun doc ->
+            match doc#v with
+            | Verbatim code -> string code
+            | Expr e -> self#print_expr AstPosition_Quote e
+            | Pattern p -> self#print_pat AstPosition_Quote p
+            | Typ t -> self#print_ty AstPosition_Quote t)
```

Comment:
Here, we should directly use `doc#p`: this will call into the underlying methods `self#quote_content_{Verbatim,Expr,Pattern,Typ}`